cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -348,6 +348,10 @@ protected void startUp() throws Exception {
         Transactions.invalidateQuietly(txClient, tx);
         throw t;
       }
+    } catch (LinkageError e) {
+      // Need to wrap LinkageError. Otherwise, listeners of this Guava Service may not be called if the initialization
+      // of the user program is missing dependencies (CDAP-2543)
```

Comment:
It seems like destroy doesn't need it, because it already happens in the shutDown method of the Service.
The issue if in `run` already is handled by being wrapped into RuntimeException by the drivers/services.